src/chat/runeMagicCard.ts

```diff
@@ -0,0 +1,411 @@
+import { ItemDataSource } from "@league-of-foundry-developers/foundry-vtt-types/src/foundry/common/data/data.mjs/itemData";
+import { Rune } from "../actors/item-specific/rune";
+import { RqgActor } from "../actors/rqgActor";
+import { RqgActorDataSource } from "../data-model/actor-data/rqgActorData";
+import { CultDataSource } from "../data-model/item-data/cultData";
+import { ItemTypeEnum, RqgItemDataSource } from "../data-model/item-data/itemTypes";
+import { RuneDataSource } from "../data-model/item-data/runeData";
+import { Ability, ResultEnum, ResultMessage } from "../data-model/shared/ability";
+import { RqgItem } from "../items/rqgItem";
+import {
+  assertItemType,
+  getActorFromIds,
+  getGame,
+  getJournalEntryName,
+  getSpeakerName,
+  RqgError,
+  usersThatOwnActor,
+} from "../system/util";
+
+type ShortRune = {
+  name: string;
+  id: string;
+  chance: number;
+};
+
+type RuneMagicCardFlags = {
+  actorId: string;
+  tokenId: string | null; // Needed to avoid saving a full (possibly syntetic) actor
+  itemData: ItemDataSource;
+  eligibleRunes: RuneDataSource[];
+  formData: {
+    runePointCost: number;
+    cultId: string;
+    magicPointBoost: number;
+    ritualOrMeditation: number;
+    skillAugmentation: number;
+    otherModifiers: number;
+    chance: number;
+    selectedRuneId: string;
+    journalEntryName: string;
+    journalPack: string;
+    ritualOrMeditationOptions: any;
+    skillAugmentationOptions: any;
+  };
+};
+
+export class RuneMagicCard {
+  public static async show(
+    runeMagicItemId: string,
+    actor: RqgActor,
+    token: TokenDocument | null
+  ): Promise<void> {
+    const runeMagicItem = actor.items.get(runeMagicItemId);
+    assertItemType(runeMagicItem?.data.type, ItemTypeEnum.RuneMagic);
+    const cult = actor.items.get(runeMagicItem.data.data.cultId);
+    assertItemType(cult?.data.type, ItemTypeEnum.Cult);
+
+    let usableRuneNames: string[] = [];
+    let runesForCasting: RuneDataSource[] = [];
+    if (runeMagicItem.data.data.runes.includes(getGame().settings.get("rqg", "magicRuneName") as string)) {
+      // Actor can use any of the cult's runes to cast
+      // And some cults have the same rune more than once, so de-dupe them
+      usableRuneNames = [...new Set(cult.data.data.runes)];
+    } else {
+      // Actor can use any of the Rune Magic Spell's runes to cast
+      usableRuneNames = [...new Set(runeMagicItem.data.data.runes)];
+    }
+
+    // Get the actor's versions of the runes, which will have their "chance"
+    usableRuneNames.forEach((rune) => {
+      //@ts-ignore name
+      const actorRune = actor.items.getName(rune);
+      assertItemType(actorRune?.data.type, ItemTypeEnum.Rune);
+      runesForCasting.push(actorRune.data);
+    });
+
+    const strongestRune = runesForCasting.reduce(function (prev, current) {
+      //@ts-ignore data WHY?!
```

Comment:
Don't think this ts-ignore is needed?

Reply:
Weird.  Because I definitely double-checked all the ts-ignores and couldn't get rid of them before, but now I can.

Reply:
same